experimental/inductor.py

```diff
@@ -0,0 +1,218 @@
+        #    - We collect all concrete getattr inputs into "attr_vals"
+        from collections import defaultdict
+        attr_vals_map = dict()
+        attr_nodes = defaultdict(lambda: [])
+
+        for p in k2.graph.nodes:
+            if p.op == 'get_attr':
+                val = getattr(k2, p.target)
+                attr_vals_map[p.target] = val
+                attr_nodes[p.target].append(p)
+
+        after_last_placeholder = None
+        for i, p in enumerate(k2.graph.nodes):
+            if p.op != 'placeholder':
+                after_last_placeholder = p
+                break
+
+        attr_vals = []
+        with k2.graph.inserting_before(after_last_placeholder):
+            for attr, nodes in attr_nodes.items():
+                placeholder = k2.graph.placeholder(attr)
+                attr_vals.append(attr_vals_map[attr])
+                for node in nodes:
+                    node.replace_all_uses_with(placeholder)
+                    k2.graph.erase_node(node)
+    
+
+        # 3. Remove unused inputs
+        #    - This is needed in order to remove non-tensor inputs before passing this
+        #      graph to inductor. Inductor expects only tensors as inputs
+        #    - This works since all the non-tensor inputs should be unused at this point,
+        #      since in principle we should have converted all getattrs into inputs.
+        #    - We keep the indices of unused inputs in "unused_inputs_idx"
+        unused_inputs = []
+        unused_inputs_idx = []
+
+        for i, p in enumerate(k2.graph.nodes):
+            if p.op != 'placeholder':
+                break
+            if len(p.users) == 0:
+                unused_inputs.append(p)
+                unused_inputs_idx.append(i)
+
+        for p in unused_inputs:
+            k2.graph.erase_node(p)
+
+        k2.recompile()
+
+ 
+        # 4. call make_fx a second time with functionalization
+        with torch.no_grad():
+            used_args_and_attrs = prepare_args(attr_vals, unused_inputs_idx, args)
+            k3 = make_fx(functionalize(k2), decomposition_table=all_decomps)
+            k4 = k3(*used_args_and_attrs, **kwargs)
+
+        for p in k4.graph.nodes:
+            if p.target == torch.ops.aten.alias.default:
+                p.replace_all_uses_with(p.args[0])
+
+        torch.fx.node._side_effectful_functions.add(torch.ops.aten.copy_.default)
+        k4.graph.eliminate_dead_code()
+        k4.recompile()
+
+        return k4, attr_vals, unused_inputs_idx
+
+    return call
+
+
+def make_inductor_fn(fn, example_inputs):
+    """
+    Use inductor to compile the function fn. 
+    Uses make_functional_fx to trace it passing example_inputs.
+    """
+    fnfx, attr_vals, unused_input_idx = make_functional_fx(fn)(*example_inputs)
+
+    used_args = prepare_args(attr_vals, unused_input_idx, example_inputs)
+    compiled_f = compile_fx_inner(fnfx, used_args)
```

Comment:
why compile_fx_inner vs compile_fx?

Reply:
compile_fx calls aot_autograd. in this example I bypass aot_autograd , replacing it with make_functional_fx above.